Vagrantfile

```diff
@@ -379,8 +381,8 @@ Vagrant.configure('2') do |config|
             end
 
             # Set hostname
-            if p.key?('set_hostname')
-                if p.key?('hostname')
+            if param(p, :set_hostname) or p.key?(:hostname)
```

Comment:
I agree with this condition although you should be using `p.key?(:set_hostname)` instead of `params(p, :set_hostname)`. Also the logic would be better to write like this:

```ruby
if p.key?(:hostname)
    node.vm.hostname = param(p, :hostname)
elsif p.key?(:set_hostname)
    node.vm.hostname = name
```

Reply:
Setting this `p.key?(:set_hostname)` instead of `params(p, :set_hostname)` will not allow to set default `set_hostname`
There is better way to check boolean in ruby that `elsif param(p, :set_hostname)`?

```ruby

            if p.key?(:hostname)
                node.vm.hostname = p[:hostname]
            elsif param(p, :set_hostname)
                node.vm.hostname = name
            end
```

Reply:
Yes, you are right. It makes sense to set `set_hostname` in `defaults` as well so `param()` should be used here.